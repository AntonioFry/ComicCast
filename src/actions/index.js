export const favoriteCharacter = (characterId) => ({
  type: "FAVORITE_CHARACTER",
  characterId
})