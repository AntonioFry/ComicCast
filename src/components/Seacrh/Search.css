.search-form {
  margin: 20px;
  display: flex;
  justify-content: center;
}

.character-search-input {
  font-size: 1.75em;
  height: 40px;
  width: 80%;
}