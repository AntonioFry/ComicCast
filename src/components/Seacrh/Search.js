import React, { Component } from 'react';

class Search extends Component {
  constructor() {
    super();
    this.state = {
      search: '',
    }
  }

  handleChange = (e) => {

  }

  handleSubmit = (e) => {

  }

  render() {
    return (
      <form>
        <input
          className="character-search-input"
          placeholder="Search for a character"
          value={this.state.search}
          name="search"
        />
      </form>
    )
  }
}

export default Search;