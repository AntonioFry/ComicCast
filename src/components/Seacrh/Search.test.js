import React from 'react';
import { Search } from './Search';
import { shallow } from 'enzyme';
import { setSearchResult } from '../../actions';
import { mapDispatchToProps } from './Search';

describe('Search', () => {

  describe("Seacrh component", () => {
    let wrapper;
    let e;
  
    beforeEach(() => {
      wrapper = shallow(<Search />)
      e = {
        target: {
          name: 'search',
          value: 'something'
        }
      }
    });
  
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  
    it('should change state when handleChange is invoked', () => {
      wrapper.instance().handleChange(e);
      expect(wrapper.state('search')).toEqual('something');
    })
  
    it('should invoke handleChange on keyUp in the input', () => {
      wrapper.instance().handleChange = jest.fn();
      wrapper.find('.character-search-input').simulate('change', { value: 'something' });
      expect(wrapper.instance().handleChange).toHaveBeenCalled();
    });
  
    it('should invoke handleSubmit when handleChange is invoked', async () => {
      wrapper.instance().handleSubmit = jest.fn();
      await wrapper.instance().handleChange(e);
      expect(wrapper.instance().handleSubmit).toHaveBeenCalled();
    });
  });

  describe('mapDispatchToProps', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = setSearchResult("search")
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.setSearchResult("search");
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});