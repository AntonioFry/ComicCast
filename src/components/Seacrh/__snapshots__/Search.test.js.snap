// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Seacrh component should match the snapshot 1`] = `
<form
  className="search-form"
>
  <input
    className="character-search-input"
    name="search"
    onChange={[Function]}
    placeholder="Search for a character"
    value=""
  />
</form>
`;
